docs/source/cost.rst

```diff
@@ -0,0 +1,329 @@
+number. The `net` parameter is a boolean that indicates if the
+aggregation should keep the sign of the error, or take the absolute
+value of the error before aggregating. Note that when :code:`net ==
+True` and the cost per unit error is positive, it is possible to
+calculate a final cost that is negative.
+
+
+Basic Cost Models
+=================
+
+Constant
+--------
+
+The constant cost model parameters are defined by
+:py:class:`.ConstantCost` and implemented for deterministic forecasts
+by :py:func:`solarforecastarbiter.metrics.deterministic.constant_cost`.
+This model expects a `cost` parameter with units of $ per unit error.
+Thus, if one were comparing an AC power forecasts to observations, this
+cost would be assumed to have units of $/MW. One could scale this cost
+based on the forecast interval length to mimic a cost per MWh.
+
+:py:class:`.ConstantCost` also expects `aggregation` and `net`
+parameters. `aggregation` defines how costs are aggregated for a given
+analysis time period (as specified in the report). Options include
+`sum` and `mean`. The `net` parameter is boolean that specifies
+whether or not the the sum/mean should be taken without (True) or with
+(False) the absolute value of the error.
+
+An example of a cost that would take the mean of error values after
+taking the absolute value and applying a cost of $2.5/unit error is
+
+.. code-block:: python
+
+    from solarforecastarbiter import datamodel
+
+    cost_model = solarforecastarbiter.datamodel.ConstantCost(
+        cost=2.5,
+        aggregation='mean',
+        net=False
+    )
+
+
+Time of Day
+-----------
+
+The time-of-day cost model parameters are defined using
+:py:class:`.TimeOfDayCost` and implemented for deterministic forecasts
+by
+:py:func:`solarforecastarbiter.metrics.deterministic.time_of_day_cost`.
+Similar to the constant cost, the datamodel expects `aggregation` and
+`net` parameters. In this case, `cost` is an iterable of cost values
+that are paired with each time given by `times`. The `fill` parameter
+specifies how the costs should be extended to times that are not
+included in `times`. Options for `fill` include 'forward' and
+'backward'. This filling parameter also controls how values "wrap
+around" midnight. For example, for a cost describing different costs
+depending on a evening peak,
+
+.. code-block:: python
+
+    import datetime
+    from solarforecastarbiter import datamodel
+
+    cost_model = datamodel.TimeOfDayCost(
+        cost=[3.3, 1.2],
+        times=[datetime.time(hour=15), datetime.time(hour=20)],
+        net=True,
+        aggregation='sum',
+        fill='forward',
+    )
+
+the value of $3.3 / unit error applies from 15:00 to just before
+20:00, and the value of $1.2 / unit error applies for all other times
+in the day *except* 15:00 to 20:00. The `timezone` parameter defines
+the timezone the `times` are referenced in. If `timezone` is None,
+`times` is assumed to be in same timezone as the errors.
```

Comment:
In my mind, the cost is quite different from forecast/obs and using interval to describe it confuses the interval idea. For forecast/obs the interval describes a period to apply some statistic to. The cost is just some function. I don't see a relation between the forecast interval label and the cost. 

If cost has an interval label, it should be defined separately. We don't want to try and compute the same cost for different forecasts but get inconsistent answers because the interval labels are different.